Sort kubectl top pod output when --sort-by and --containers are used together

brianpursley opened the pull request:
- Changed kubectl top pod to sort first at the pod level and then at the container level when --sort-by and --containers are used together.
- Refactored printSinglePodMetrics into two separate functions instead of passing in bool to change behavior.
- Refactored MetricsSorters to simplify code.
- Added unit tests to test container sorting.
- Fixed pod sorting unit tests which were not working because it was checking for --sort-by command line flag which was never true.

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
When running `kubectl top pod --containers --sort-by=cpu` (or `--sort-by=memory`), the output is unsorted.

This PR fixes the problem by sorting the output first at the pod-level, and then at the container-level when `--sort-by` and `--containers` are used together.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubectl/issues/901

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Fixed bug where kubectl top pod output is not sorted when --sort-by and --containers flags are used together
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

brianpursley commented:
/assign @apelisse

apelisse commented on staging/src/k8s.io/kubectl/pkg/metricsutil/metrics_printer.go:
```diff
@@ -144,7 +144,42 @@ func NewPodMetricsSorter(metrics []metricsapi.PodMetrics, printContainers bool,
+	}
+}
+
+type ContainerMetricsSorter struct {
```
Use the Slice function? with two/three separate sorting functions

apelisse reviewed (commented):
I know very little about this code so I would strongly suggest another pair of eyes to look at it

brianpursley commented:
> I know very little about this code so I would strongly suggest another pair of eyes to look at it

Ok, thanks for taking a look, I will look back in history to see who approved previous PRs involving this code.

/unassign apelisse

brianpursley committed:
Sort kubectl top output when --sort-by and --containers are used together

- Changed kubectl top to sort first at the pod level and then at the container level when --sort-by and --containers are used together.
- Refactored printSinglePodMetrics into two separate functions instead of passing in bool to change behavior.
- Refactored MetricsSorters to simplify code.
- Added unit tests to test container sorting.
- Fixed pod sorting unit tests which were not working because it was checking for --sort-by command line flag which was never true.

brianpursley commented:
/assign @soltysh

brianpursley commented:
/retest

brianpursley commented:
/retest

soltysh reviewed (approved):
/lgtm
/approve
/milestone v1.20
/priority backlog

soltysh commented:
/refresh all
